sdk/azidentity/chained_token_credential.go

```diff
@@ -19,7 +19,8 @@ type ChainedTokenCredentialOptions struct {
 
 // ChainedTokenCredential is a chain of credentials that enables fallback behavior when a credential can't authenticate.
 type ChainedTokenCredential struct {
-	sources []azcore.TokenCredential
+	sources              []azcore.TokenCredential
+	successfulCredential *azcore.TokenCredential
```

Comment:
I don't think this has to be an explicit pointer, interfaces are pointers under the hood.

Reply:
ok ok! thank you!